healthcheck/healthcheck_test.go

```diff
@@ -193,6 +193,64 @@ func TestSetBackendsConfiguration(t *testing.T) {
 	}
 }
 
+func TestNewRequest(t *testing.T) {
+	tests := []struct {
+		host     string
+		port     int
+		path     string
+		expected string
+	}{
+		{
+			host:     "backend1:80",
+			port:     0,
+			path:     "/test",
+			expected: "http://backend1:80/test",
+		},
+		{
+			host:     "backend2:80",
+			port:     8080,
+			path:     "/test",
+			expected: "http://backend2:8080/test",
+		},
+		{
+			host:     "backend1",
+			port:     0,
+			path:     "/health",
+			expected: "http://backend1/health",
+		},
+		{
+			host:     "backend2",
+			port:     8080,
+			path:     "/health",
+			expected: "http://backend2:8080/health",
+		},
+	}
+
+	for _, test := range tests {
+		backend := NewBackendHealthCheck(
+			Options{
+				Path: test.path,
+				Port: test.port,
+			})
+
+		u := &url.URL{
+			Scheme: "http",
+			Host:   test.host,
+		}
+
+		req, err := backend.newRequest(u)
+
+		if err != nil {
+			t.Errorf("failed to create new backend request for %s", u.String())
+		}
+
+		actual := req.URL.String()
+		if test.expected != actual {
+			t.Errorf("got %s for healthcheck URL, wanted %s", actual, test.expected)
```

Comment:
Include test-case identifiers here too.

Since we need it at two spots, it seems reasonable to me to extract them into a variable.